server/apps/distributed-app/docs/modules/ROOT/pages/configure/imap.adoc

```diff
@@ -95,4 +107,11 @@ will use the specified value.
 | connectionLimitPerIP
 | Set the maximum simultaneous incoming connections per IP for this service
 
-|===
\ No newline at end of file
+|===
+
+=== OIDC setup
+James IMAP support XOAUTH2 authentication mechanism which allow authenticating against a OIDC providers.
```

Comment:
Opened question: I'm starting wondering if we should not review our vocabulary usage here (and our code) between OAuth2 and OIDC.

OIDC is something on top of OAuth2, with the SSO concept added for example. Here with IMAP/SMTP it feels like we are only doing OAuth2 though (just getting a token... not really any SSO/SLO concepts with IMAP clients so far).

I feel like we are mixing everything up together sometimes. Like in imap we have the conf under a oauth block, while in smtp it's under a oidc block, etc.

IDK what do you think guys?

Reply:
> Like in imap we have the conf under a oauth block, while in smtp it's under a oidc block, etc.

If that's the case then of course we should normalize such choices and only keep one coherent configuration parameters.

(please open a ticket in the Linagora board to do so, so that we do not forget.

> OAUTH2 VS OIDC semantic

We did implement OIDC (which is a sub-protocol of OAUTH2) and we actually actively rely on it as we advertize the OIDC config endpoint upon auth failure (thus expect the clients to recognize it!).

Strict OAUTH2 usage would require extra client config (URL of the OIDC provider) to bypass this OIDC detail and stick to strict OAUTH2.

Thus I am more in favor of the OIDC wording. 

My 2 cents.

Reply:
> Strict OAUTH2 usage would require extra client config (URL of the OIDC provider) to bypass this OIDC detail and stick to strict OAUTH2.

That's pretty much what we needed to do to config TB... But I think we agree to say that the client should recognize it indeed but doesn't. So ok :)

> Thus I am more in favor of the OIDC wording.

Said like this makes sense, I'm ok.

> (please open a ticket in the Linagora board to do so, so that we do not forget.)

Sure :)

Reply:
=> https://github.com/linagora/james-project/issues/4467